sklearn/tests/test_calibration.py

```diff
@@ -192,27 +219,30 @@ def test_calibration_multiclass():
     X_test, y_test = X[1::2], y[1::2]
 
     clf.fit(X_train, y_train)
-    for method in ['isotonic', 'sigmoid']:
-        cal_clf = CalibratedClassifierCV(clf, method=method, cv=2)
-        cal_clf.fit(X_train, y_train)
-        probas = cal_clf.predict_proba(X_test)
-        assert_array_almost_equal(np.sum(probas, axis=1), np.ones(len(X_test)))
-
-        # Check that log-loss of calibrated classifier is smaller than
-        # log-loss of naively turned OvR decision function to probabilities
-        # via softmax
-        def softmax(y_pred):
-            e = np.exp(-y_pred)
-            return e / e.sum(axis=1).reshape(-1, 1)
-
-        uncalibrated_log_loss = \
-            log_loss(y_test, softmax(clf.decision_function(X_test)))
-        calibrated_log_loss = log_loss(y_test, probas)
-        assert uncalibrated_log_loss >= calibrated_log_loss
+
+    cal_clf = CalibratedClassifierCV(
+        clf, method=method, cv=2, ensemble=ensemble
+    )
+    cal_clf.fit(X_train, y_train)
+    probas = cal_clf.predict_proba(X_test)
+    # Check probabilities sum to 1
+    assert_array_almost_equal(np.sum(probas, axis=1), np.ones(len(X_test)))
+
+    # Check that log-loss of calibrated classifier is smaller than
+    # log-loss obtained by naively turning OvR decision function to
+    # probabilities via softmax
+    def softmax(y_pred):
```

Comment:
there is a function defines in `sklearn.utils.extmath` already. can we use it?

Reply:
The `extmath` fun seems to normalise:
```Python
    if copy:
        X = np.copy(X)
    max_prob = np.max(X, axis=1).reshape((-1, 1))
    X -= max_prob
    np.exp(X, X)
    sum_prob = np.sum(X, axis=1).reshape((-1, 1))
    X /= sum_prob
    return X
```
which I don't think is the same as the softmax fun defined here? Regardless the test fails if I change it...